src/transformers/modeling_utils.py

```diff
@@ -3207,9 +3208,38 @@ def from_pretrained(
                         )
                         if resolved_archive_file is not None:
                             is_sharded = True
-                    if resolved_archive_file is None:
-                        # Otherwise, maybe there is a TF or Flax model file.  We try those to give a helpful error
-                        # message.
+
+                    if resolved_archive_file is not None:
+                        # If the PyTorch file was found, check if there is a safetensors file on the repository
+                        # If there is no safetensors file on the repositories, start an auto conversion
+                        safe_weights_name = SAFE_WEIGHTS_INDEX_NAME if is_sharded else SAFE_WEIGHTS_NAME
+                        has_file_kwargs = {
+                            "revision": revision,
+                            "proxies": proxies,
+                            "token": token,
+                        }
+                        cached_file_kwargs = {
+                            "cache_dir": cache_dir,
+                            "force_download": force_download,
+                            "resume_download": resume_download,
+                            "local_files_only": local_files_only,
+                            "user_agent": user_agent,
+                            "subfolder": subfolder,
+                            "_raise_exceptions_for_gated_repo": False,
+                            "_raise_exceptions_for_missing_entries": False,
+                            "_commit_hash": commit_hash,
+                            **has_file_kwargs,
+                        }
+                        if not has_file(pretrained_model_name_or_path, safe_weights_name, **has_file_kwargs):
+                            cls._auto_conversion = Thread(
+                                target=auto_conversion,
+                                args=(pretrained_model_name_or_path,),
+                                kwargs=cached_file_kwargs,
+                            )
+                            cls._auto_conversion.start()
```

Comment:
@LysandreJik I'm not shocked by having a `cls._auto_conversion` attribute TBH. Though a solution to get rid of it is to give a name to the thread. Something like that:

```py
Thread(
    target=auto_conversion,
    args=(pretrained_model_name_or_path,),
    kwargs=cached_file_kwargs,
    name="Thread-autoconversion-{<unique id here>}",
).start()
```

and then in the tests:
```py
for thread in threading.enumerate():
    print(thread.name)
# ...
# Thread-autoconversion-0
```

Thread names don't have to be unique BTW (they have a thread id anyway). But I think it's best to at least assign a unique number to the name.

Reply:
But it's quite hacky IMO. In a simple case it should work fine but if you start to have several threads / parallel tests, it might get harder to be 100% sure the thread you've started is indeed the one you retrieve in the test logic.

Reply:
yeah here it's really only for testing and I don't want to depend on a flaky `time.sleep` or something so ensuring that the thread joins first is optimal. The thread name is actually much better IMO, I'll implement that! Thanks a lot!